src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -955,33 +955,95 @@ private IEnumerable<Expression> CompileInvocationArguments(IEnumerable<Expressio
             return arguments == null ? Array.Empty<Expression>() : arguments.Select(CompileExpressionOperand);
         }
 
-        internal Expression ReduceAssignment(ISupportsAssignment left, TokenKind tokenKind, Expression right)
+        internal Expression ReduceAssignment(
+            ISupportsAssignment left,
+            TokenKind tokenKind,
+            Expression right,
+            MemberExpressionAst lhsMemberAccessAst)
         {
-            IAssignableValue av = left.GetAssignableValue();
-            ExpressionType et = ExpressionType.Extension;
+            IAssignableValue assignmentTarget = left.GetAssignableValue();
 
-            switch (tokenKind)
+            if (tokenKind == TokenKind.Equals)
             {
-                case TokenKind.Equals: return av.SetValue(this, right);
-                case TokenKind.PlusEquals: et = ExpressionType.Add; break;
-                case TokenKind.MinusEquals: et = ExpressionType.Subtract; break;
-                case TokenKind.MultiplyEquals: et = ExpressionType.Multiply; break;
-                case TokenKind.DivideEquals: et = ExpressionType.Divide; break;
-                case TokenKind.RemainderEquals: et = ExpressionType.Modulo; break;
-                case TokenKind.QuestionQuestionEquals: et = ExpressionType.Coalesce; break;
+                return assignmentTarget.SetValue(this, right);
             }
 
+            var expressionType = tokenKind switch
+            {
+                TokenKind.PlusEquals => ExpressionType.Add,
+                TokenKind.MinusEquals => ExpressionType.Subtract,
+                TokenKind.MultiplyEquals => ExpressionType.Multiply,
+                TokenKind.DivideEquals => ExpressionType.Divide,
+                TokenKind.RemainderEquals => ExpressionType.Modulo,
+                TokenKind.QuestionQuestionEquals => ExpressionType.Coalesce,
+                _ => ExpressionType.Extension
+            };
+
+            return ReduceCompoundAssignment(this, assignmentTarget, lhsMemberAccessAst, expressionType, right);
+        }
+
+        internal static BlockExpression ReduceCompoundAssignment(
+            Compiler compiler,
```

Comment:
Any reason to make this static, if we just have to pass `this` back in?

Reply:
Not anymore! I wrote it this way initially as I wasn't sure if I'd need to callback to this from the binder or some such witchcraft, but that's definitely not required here.